crates/bevy_ui/src/accessibility.rs

```diff
@@ -0,0 +1,157 @@
+use bevy_a11y::{
+    accesskit::{NodeBuilder, Rect, Role},
+    AccessibilityNode,
+};
+use bevy_app::{App, Plugin};
+
+use bevy_ecs::{
+    prelude::Entity,
+    query::{Changed, Or, Without},
+    system::{Commands, Query},
+};
+use bevy_hierarchy::Children;
+
+use bevy_render::prelude::Camera;
+use bevy_text::Text;
+use bevy_transform::prelude::GlobalTransform;
+
+use crate::{
+    prelude::{Button, Label},
+    Node, UiImage,
+};
+
+fn calc_name(texts: &Query<&Text>, children: &Children) -> Option<Box<str>> {
```

Comment:
Mainly because I didn't want to take on fixing bevy_ui's shortcomings in this PR. I needed something to test with, and we don't currently have a way of describing UI widgets in any form. So I calculated a name/title and left it there.

Reply:
I see. And so the shortcoming you're talking about is that bevy_ui doesn't enforce `Name` components to extract the a11y name from?

If it did, would this function instead be used to extract the node's description?

Reply:
It's more that bevy_ui will eventually need something like Flutter's [Semantics class](https://api.flutter.dev/flutter/widgets/Semantics-class.html) to truly be accessible. Ideally, in that situation, we'll have clearer understandings of how and when we're able to auto-generate names/descriptions, the Semantics component will be nicely integrated with the editor so accessibility properties can be set directly, etc. And I'm trying to take a lighter touch with all those things right now because I don't know what the right answer is.

Also, names and descriptions are very different things, and you don't necessarily want to scattershot the same property value everywhere so e.g. you don't double-announce a name because a screen reader might read a widget like "<name> <description>: button" or similar. Which properties to use and when is fairly complex and differs per widget, but in general setting names is universal and fairly low-hanging fruit. Something more complex I likely wouldn't auto-generate unless the widget semantics clearly call for it.

Ideally, we'll use AccessKit's [NodeBuilder](https://docs.rs/accesskit/latest/accesskit/struct.NodeBuilder.html) or a supported subset of its schema as the core of our Semantics component, and come to consensus about how we automatically fill in properties vs. expecting developers to fill them in themselves.

Reply:
Makes sense, thanks for the explanation!